src/store/datasets-dicom.ts

```diff
@@ -117,85 +118,104 @@ export const useDICOMStore = defineStore('dicom', {
-
-          if (!(volumeKey in this.volumeInfo)) {
-            const info = await dicomIO.readTags(volumeKey, [
-              { name: 'PatientName', tag: '0010|0010', strconv: true },
-              { name: 'PatientID', tag: '0010|0020', strconv: true },
-              { name: 'PatientBirthDate', tag: '0010|0030' },
-              { name: 'PatientSex', tag: '0010|0040' },
-              { name: 'StudyInstanceUID', tag: '0020|000d' },
-              { name: 'StudyDate', tag: '0008|0020' },
-              { name: 'StudyTime', tag: '0008|0030' },
-              { name: 'StudyID', tag: '0020|0010', strconv: true },
-              { name: 'AccessionNumber', tag: '0008|0050' },
-              { name: 'StudyDescription', tag: '0008|1030', strconv: true },
-              { name: 'Modality', tag: '0008|0060' },
-              { name: 'SeriesInstanceUID', tag: '0020|000e' },
-              { name: 'SeriesNumber', tag: '0020|0011' },
-              { name: 'SeriesDescription', tag: '0008|103e', strconv: true },
-            ]);
-
-            // TODO parse the raw string values
-            const patient = {
-              PatientID: info.PatientID || ANONYMOUS_PATIENT_ID,
-              PatientName: info.PatientName || ANONYMOUS_PATIENT,
-              PatientBirthDate: info.PatientBirthDate || '',
-              PatientSex: info.PatientSex || '',
-            };
-            const patientKey = genSynPatientKey(patient);
-
-            const studyKey = info.StudyInstanceUID;
-            const study = pick(
-              info,
-              'StudyID',
-              'StudyInstanceUID',
-              'StudyDate',
-              'StudyTime',
-              'AccessionNumber',
-              'StudyDescription'
-            );
-
-            const volumeInfo = {
-              ...pick(
-                info,
-                'Modality',
-                'SeriesInstanceUID',
-                'SeriesNumber',
-                'SeriesDescription'
-              ),
-              NumberOfSlices: numberOfSlices,
-              VolumeID: volumeKey,
-            };
-
-            updatedVolumeKeys.push({
-              patientKey,
-              studyKey,
-              volumeKey,
+        Object.entries(volumesToFilesMap).map(
+          async ([volumeKey, fileNames]) => {
+            const volumeFiles: File[] = [];
+            fileNames.forEach((fileName) => {
+              const file = nameToFileMap.get(fileName);
+              if (file) {
+                volumeFiles.push(file);
+              }
             });
 
-            this._updateDatabase(patient, study, volumeInfo);
-          }
-
-          // invalidate any existing volume
-          if (volumeKey in this.volumeToImageID) {
-            // buildVolume requestor uses this as a rebuild hint
-            set(this.needsRebuild, volumeKey, true);
+            // Add file entries to the file store
+            fileStore.add(volumeKey, volumeFiles);
```

Comment:
In `deleteVolume`, we want to have the corresponding remove operation in the fileStore.

Reply:
That is done [here](https://github.com/cjh1/paraview-medical/blob/dicom-db/src/store/datasets.ts#L259).

Reply:
Oh, I see now. Thanks!